src/modules/ModerationModule.cpp

```diff
@@ -1,32 +1,55 @@
 #include "ModerationModule.h"
 #include "UmikoBot.h"
+#include "core/Permissions.h"
 
 using namespace Discord;
 
 ModerationModule::ModerationModule()
-    : Module("moderation", true)
+	: Module("moderation", true)
 {
-    RegisterCommand(Commands::MODERATION_INVITATION_TOGGLE, "invitations",
+	RegisterCommand(Commands::MODERATION_INVITATION_TOGGLE, "invitations",
 		[this](Client& client, const Message& message, const Channel& channel)
-    {
-		m_invitationModeration ^= true;
-		client.createMessage(message.channelId(), m_invitationModeration ? "Invitations will be deleted!" : "Invitations won't be deleted!");
-    });
+	{
+			QStringList args = message.content().split(' ');
+			UmikoBot::VerifyAndRunAdminCmd(client, message, channel, 1, args, true, [this, &client, channel, message, args]()
+			{
+				m_invitationModeration ^= true;
+				client.createMessage(message.channelId(), m_invitationModeration ? "Invitations will be deleted!" : "Invitations won't be deleted!");
+			});
+	});
 }
 
 void ModerationModule::OnMessage(Client& client, const Message& message)
 {
+	client.getChannel(message.channelId()).then(
+	[this, message, &client](const Channel& channel)
+	{
+		if (m_invitationModeration)
+		{
+			if (message.content().contains("https://discord.gg/", Qt::CaseInsensitive))
+			{
+				auto authorID = message.author().id();
+				::Permissions::ContainsPermission(client, channel.guildId(), message.author().id(), CommandPermission::ADMIN,
+					[this, message, &client, authorID](bool result)
+				{
+					if (!result)
+					{
+						client.deleteMessage(message.channelId(), message.id());
+						client.createMessage(message.channelId(), "**<@" + QString::number(authorID) + ">, invitation link of servers aren't allowed in any channels on this server. Please take it to DMs!**");
```

Comment:
I would love it if instead the author would get a DM with the message they sent so that they have a chance to edit out the link. They might have spent some time writing the message and it would be a shame for the message to just be yeeted into the void for a small misunderstanding.

So if you send them two DMs. One explaining that the message was deleted because it contained an invite link, and another with the entire message in it 😊

Reply:
Perfectly Done 😊